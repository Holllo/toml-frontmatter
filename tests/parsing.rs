use {
  anyhow::Result,
  insta::{assert_snapshot, assert_toml_snapshot},
  serde::{Deserialize, Serialize},
};

#[derive(Debug, Deserialize, Serialize)]
struct Frontmatter {
  date: String,
}

#[test]
fn test_parsing() -> Result<()> {
  let sample = r#"
---toml
date = "2023-01-01"
---

# Some Markdown

With text!"#;

  let (toml, markdown) = toml_frontmatter::parse::<Frontmatter>(sample.trim())?;

  assert_toml_snapshot!("toml", toml);
  assert_snapshot!("markdown", markdown);

  Ok(())
}
