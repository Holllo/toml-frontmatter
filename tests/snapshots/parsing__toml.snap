---
source: tests/parsing.rs
expression: toml
---
date = '2023-01-01'
